### DISCUSSION TITLE
Dynamic rules

### DISCUSSION THREAD

**Discussion by kuimovvg** [seaside1/jrule#175] [General]
Is it possible to dynamically create rules from code? If not, what do you think about this possibility?

**Comment by seaside1** [comment]
I actually think you would be able to do it, since there are no restriction on using the JRule core-classes. But if it is something you would want to do, maybe it would make sense to have a more available api / helper classes for it.
What is your use-case for doing this?

I have thought about adding functionality for dynamically create items, the use-case being deploying a software "module/jar" containing both rules and dynamic items. You could then use the items directly without having to define them yourself for something specific (maybe a automated mailbox delivering notifications when you get a mail or something else).
 It would be like deploying a binding but you would not only get things, you would also automatically get items. However I'm still a huge fan of having all my items in files, so I have not developed the though any further.

**Comment by seaside1** [comment]
Take a look at this:
https://github.com/seaside1/jrule-sonos

Here I dynamically generate items and rules

**Comment by kuimovvg** [comment]
This functionality is implemented in the issue https://github.com/seaside1/jrule/pull/177
